src/Engine/EmitMSIL/Replication.cs

```diff
@@ -556,7 +555,6 @@ private static object ExecWithZeroRI(CLRFunctionEndPoint finalFep, List<CLRStack
             }
 
             var returnVal = marshaller.UnMarshal(dsRetValue, finalFep.ReturnType, runtimeCore);
-            marshaller.OnDispose(dsRetValue);
```

Comment:
I don't know why I added these dispose calls. They are not in the old VM.

Reply:
hmm - we also don't marshal back and forth as frequently in the old vm, do we?
Do we see any memory use increase after this pr?

Reply:
I can check it...but there is a task to stop all the extra marshaling and expose the CLRStackValue to the rest of the new VM (not just runtime replication)